flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/com/ververica/cdc/connectors/starrocks/sink/EventRecordSerializationSchema.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.starrocks.sink;
+
+import com.starrocks.connector.flink.table.data.DefaultStarRocksRowData;
+import com.starrocks.connector.flink.table.data.StarRocksRowData;
+import com.starrocks.connector.flink.table.sink.v2.RecordSerializationSchema;
+import com.starrocks.connector.flink.tools.JsonWrapper;
+import com.ververica.cdc.common.data.RecordData;
+import com.ververica.cdc.common.event.CreateTableEvent;
+import com.ververica.cdc.common.event.DataChangeEvent;
+import com.ververica.cdc.common.event.Event;
+import com.ververica.cdc.common.event.SchemaChangeEvent;
+import com.ververica.cdc.common.event.TableId;
+import com.ververica.cdc.common.schema.Column;
+import com.ververica.cdc.common.schema.Schema;
+import com.ververica.cdc.common.utils.Preconditions;
+import com.ververica.cdc.common.utils.SchemaUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.ververica.cdc.connectors.starrocks.sink.StarRocksUtils.createFieldGetter;
+
+/** Serializer for the input {@link Event}. It will serialize a row to a json string. */
+public class EventRecordSerializationSchema implements RecordSerializationSchema<Event> {
+
+    private static final long serialVersionUID = 1L;
+
+    /** keep the relationship of TableId and table information. */
+    private transient Map<TableId, TableInfo> tableInfoMap;
+
+    private transient DefaultStarRocksRowData reusableRowData;
+    private transient JsonWrapper jsonWrapper;
+
+    @Override
+    public void open() {
+        this.tableInfoMap = new HashMap<>();
+        this.reusableRowData = new DefaultStarRocksRowData();
+        this.jsonWrapper = new JsonWrapper();
+    }
+
+    @Override
+    public StarRocksRowData serialize(Event record) {
+        if (record instanceof SchemaChangeEvent) {
+            applySchemaChangeEvent((SchemaChangeEvent) record);
+            return null;
```

Comment:
Just a worry, Will this lead to NoPointException?

Reply:
Thanks for the reminding. Connector will deal with the [null value](https://github.com/StarRocks/starrocks-connector-for-apache-flink/blob/main/src/main/java/com/starrocks/connector/flink/table/sink/v2/StarRocksWriter.java#L140).